cg/store/api/status.py

```diff
@@ -629,34 +645,43 @@ def analyses_to_deliver(self, pipeline: Pipeline = None) -> Query:
         )
 
     def analyses_to_delivery_report(self, pipeline: Pipeline = None) -> Query:
-        """Fetch analyses that needs the delivery report to be regenerated."""
+        """Fetches analyses that need a delivery report to be regenerated"""
 
-        analyses_query = self.latest_analyses()
+        records = self.Analysis.query.join(models.Analysis.family)
 
-        analyses_query = (
-            analyses_query.filter(models.Analysis.uploaded_at)
-            .filter(VALID_DATA_IN_PRODUCTION < models.Analysis.started_at)
-            .join(models.Family, models.Family.links, models.FamilySample.sample)
-            .filter(
-                and_(
-                    models.Analysis.pipeline == str(pipeline),
-                    models.Family.data_delivery.contains(DataDelivery.SCOUT),
-                    models.Sample.delivered_at.isnot(None),
-                )
-            )
-            .filter(
-                or_(
-                    models.Analysis.delivery_report_created_at.is_(None),
-                    and_(
-                        models.Analysis.delivery_report_created_at.isnot(None),
-                        models.Analysis.delivery_report_created_at > VALID_DATA_IN_PRODUCTION,
-                        models.Analysis.delivery_report_created_at < models.Sample.delivered_at,
-                    ),
-                ),
-            )
-            .order_by(models.Analysis.uploaded_at.desc())
+        records = (
```

Comment:
It would be more readable if the records filtering was functions like "get_records_for_pipeline" or similar. Than one would instantly know the aim of each filter and it would be easier to test each individual filter. It would also allow for reuse as you are using the same code again in analyses_to_upload_delivery_reports

Reply:
Added a new function and removed the code duplication [fd30ee2](https://github.com/Clinical-Genomics/cg/pull/1536/commits/fd30ee2bc13f5f8a448b8e0797eafcc09263aa11)